docs/visualize.md

```diff
@@ -0,0 +1,100 @@
+# Installation
+
+To create state machine graphs you must install `automat` with the graphing dependencies.
+
+```
+pip install automat[visualize]
+```
+
+# Example
+
+Given the following project structure:
+
+```
+mystate/
+├── __init__.py
+└── machine.py
+```
+
+And the following state machine defined in `__init__.py`
+
+```
+from automat import MethodicalMachine
+
+class MyMachine(object):
+    _machine = MethodicalMachine()
+
+    @_machine.state(initial=True)
+    def state_a(self):
+        """
+        State A
+        """
+
+    @_machine.state()
+    def state_b(self):
+        """
+        State B
+        """
+
+    @_machine.input()
+    def change_state(self):
+        """
+        Change state
+        """
+
+    @_machine.input()
+    def output_on_change_state(self):
+        """
+        Change state
+        """
+        return "Changing state"
+
+    state_a.upon(change_state, enter=state_b, outputs=[output_on_change_state])
+```
+
+You can generate a state machine visualization by running:
+
+```
+$ automat-visualize mystate
+mystate.machine.MyMachine._machine ...discovered
+mystate.machine.MyMachine._machine ...wrote image and dot into .automat_visualize
+```
+
+
+The `dot` file and `png` will be saved in the default output directory of `.automat_visualize/mystatemachine.MyMachine._machine.dot.png
```

Comment:
Just one more thing - this needs a closing backtick (``` ` ```), and the directory is `.automat_visualize/`.